File: discord_bot/webhook_reactions/issue_event.py
import logging
import discord
from discord_bot.discord_bot import get_current_bot

ISSUE_CHANNEL_ID = 1536884393870893137
log = logging.getLogger(__name__)


async def handle_issue_event(payload) -> None:
    _ = payload
    bot = get_current_bot()

    if bot is None:
        log.error("Current bot not set, not posting issue event")
        return
    channel = bot.get_channel(ISSUE_CHANNEL_ID)
    if not isinstance(channel, discord.TextChannel):
        log.error("Could not issue channel")
        return
    issue_dict = payload["issue"]
    url = issue_dict["repository_url"]
    issue_num = issue_dict["number"]
    title = issue_dict["title"]
    created_by = issue_dict["user"]["login"]
    body = issue_dict["body"]
    body_text = body if len(body) < 1200 else "Body too long for discord, view url"

    embed = discord.Embed(
        title="New issue created",
        description=f"#{issue_num}: {title}:\n{body_text}",
        color=0x00FF00,
    )
    embed.add_field(name="Issue Url", value=url, inline=False)
    embed.add_field(name="Created by", value=created_by, inline=False)

    await channel.send(embed=embed)

File: discord_bot/webhook_reactions/issue_event.go
package webhook_reactions

import (
	"fmt"
	"log"

	"github.com/bwmarrin/discordgo"

	discordbot "sgg-discord-bot/discord_bot"
)

const IssueChannelID = "1536884393870893137"

func handleIssueCreateEvent(issueData map[string]any) *discordgo.MessageEmbed {
	url, _ := issueData["repository_url"].(string)
	issueNum, _ := issueData["number"].(float64)
	title, _ := issueData["title"].(string)
	userData, _ := issueData["user"].(map[string]any)
	createdBy, _ := userData["login"].(string)
	body, _ := issueData["body"].(string)

	bodyText := body
	if len(body) >= 1200 {
		bodyText = "Body too long for discord, view url"
	}

	embed := &discordgo.MessageEmbed{
		Title:       "New issue created",
		Description: fmt.Sprintf("#%d: %s:\n%s", int(issueNum), title, bodyText),
		Color:       0x2784F5,
		Fields: []*discordgo.MessageEmbedField{
			{Name: "Issue Url", Value: url, Inline: false},
			{Name: "Created by", Value: createdBy, Inline: false},
		},
	}
	return embed
}

func handleIssueClosedEvent(issueData map[string]any) *discordgo.MessageEmbed {
	url, _ := issueData["url"].(string)
	issueNum, _ := issueData["number"].(float64)
	title, _ := issueData["title"].(string)
	userData, _ := issueData["user"].(map[string]any)
	closedBy, _ := userData["login"].(string)

	embed := &discordgo.MessageEmbed{
		Title:       "Issue closed",
		Description: fmt.Sprintf("#%d: %s", int(issueNum), title),
		Color:       0xE727F5,
		Fields: []*discordgo.MessageEmbedField{
			{Name: "Issue Url", Value: url, Inline: false},
			{Name: "Closed by", Value: closedBy, Inline: false},
		},
	}
	return embed
}

func HandleIssueEvent(payload map[string]any) {
	bot := discordbot.GetCurrentBot()
	if bot == nil {
		log.Println("Current bot not set, not posting issue event")
		return
	}
	issueType, ok := payload["action"].(string)
	if !ok {
		log.Println("Issue type could not be determined!")
		return
	}

	issueData, ok := payload["issue"].(map[string]any)
	if !ok {
		log.Println("Could not parse issue data")
		return
	}
	var embed *discordgo.MessageEmbed
	switch issueType {
	case "closed":
		embed = handleIssueClosedEvent(issueData)
	case "opened":
		embed = handleIssueCreateEvent(issueData)
	}

	_, err := bot.ChannelMessageSendEmbed(IssueChannelID, embed)
	if err != nil {
		log.Printf("Error sending issue message: %v", err)
	}
}
